test/api-test/index.js

```diff
@@ -0,0 +1,23 @@
+'use strict'
+/* global describe */
+
+/**
```

Comment:
As a rule, I usually try to make the directory structure of the tests match the directory structure of the project.

If you would like to make a case for moving tests, I would advocate:

```
test/lib/completion.js
test/lib/integration.js
test/integration/index.js
test/yargs.js
test/yargs-async.js
```

---

However, could we hold off on moving around all the tests until a later date? It makes this pull request much harder to review because it's hard to tell what's new related to the async API, and what is just tests being moved into a new folder.